File: examples/dekita/.claude/hooks/ci_monitor_session_id_check.py
#!/usr/bin/env python3
"""ci_monitor.pyのsession-idオプション指定を検出する。

Why:
    session-idが指定されないと、ci_monitor.pyはppidベースのセッション特定に
    フォールバックし、ログがClaude Codeセッションと正しく紐付かなくなる。

What:
    - ci_monitor.py呼び出しを検出
    - --session-idオプションの有無を確認
    - 未指定の場合は警告メッセージを表示

Remarks:
    - 警告型フック（ブロックしない、警告のみ）
    - PreToolUse:Bashで発火
    - AGENTS.mdにsession-id指定が必須と記載されている

Changelog:
    - silenvx/dekita#2389: フック追加
"""

import json
import re
import sys

from lib.execution import log_hook_execution
from lib.results import make_approve_result
from lib.session import parse_hook_input


def main():
    """Entry point for the CI monitor session ID check hook."""
    try:
        input_json = parse_hook_input()
        tool_input = input_json.get("tool_input") or {}
        command = tool_input.get("command") or ""

        # Check if this is a ci_monitor.py call
        if not re.search(r"ci_monitor\.py", command):
            # Not a ci_monitor.py call, approve
            result = make_approve_result("ci-monitor-session-id-check")
            print(json.dumps(result))
            return

        # Check if --session-id is provided (supports both space and = forms)
        # --session-id abc123 or --session-id=abc123 (multiple spaces allowed)
        # Note: =\S+ requires value immediately after =, \s+\S+ allows spaces before value
        if re.search(r"--session-id(?:=\S+|\s+\S+)", command):
            # --session-id is provided, approve
            result = make_approve_result("ci-monitor-session-id-check")
            log_hook_execution("ci-monitor-session-id-check", "approve", "--session-id provided")
            print(json.dumps(result))
            return

        # ci_monitor.py called without --session-id - warn
        warning = (
            "[ci-monitor-session-id-check] 警告: --session-id が指定されていません\n\n"
            "ログが正しいセッションと紐付かなくなります。\n"
            "--session-id を追加してください:\n\n"
            "  python3 .claude/scripts/ci_monitor.py {PR} --session-id <SESSION_ID>\n\n"
            "※ <SESSION_ID>はUserPromptSubmit hookで提供されるセッションID\n"
            "  例: 3f03a042-a9ef-44a2-839a-d17badc44b0a"
        )
        result = make_approve_result("ci-monitor-session-id-check", warning)
        log_hook_execution(
            "ci-monitor-session-id-check",
            "approve_with_warning",
            "--session-id missing",
            {"command": command[:100]},
        )
        print(json.dumps(result))

    except Exception as e:
        # On error, approve to avoid blocking legitimate commands
        print(f"[ci-monitor-session-id-check] Hook error: {e}", file=sys.stderr)
        result = make_approve_result("ci-monitor-session-id-check", f"Hook error: {e}")
        log_hook_execution("ci-monitor-session-id-check", "approve", f"Hook error: {e}")
        print(json.dumps(result))


if __name__ == "__main__":
    main()

File: examples/dekita/.claude/hooks/handlers/ci_monitor_session_id_check.ts
#!/usr/bin/env bun
/**
 * ci_monitor（TypeScript版）のsession-idオプション指定を検出する。
 *
 * Why:
 *   session-idが指定されないと、ci_monitorはppidベースのセッション特定に
 *   フォールバックし、ログがClaude Codeセッションと正しく紐付かなくなる。
 *
 * What:
 *   - ci_monitor_ts/main.ts呼び出しを検出
 *   - --session-idオプションの有無を確認
 *   - 未指定の場合は警告メッセージを表示
 *
 * Remarks:
 *   - 警告型フック（ブロックしない、警告のみ）
 *   - PreToolUse:Bashで発火
 *   - AGENTS.mdにsession-id指定が必須と記載されている
 *
 * Changelog:
 *   - silenvx/dekita#2389: フック追加（Python）
 *   - silenvx/dekita#3148: TypeScriptに移行
 *   - silenvx/dekita#3294: Python版ci_monitor削除に伴いTypeScript版を検出
 */

import { formatError } from "../lib/format_error";
import { logHookExecution } from "../lib/logging";
import { makeApproveResult } from "../lib/results";
import { parseHookInput } from "../lib/session";

const HOOK_NAME = "ci-monitor-session-id-check";

function approve(): void {
  console.log(JSON.stringify(makeApproveResult(HOOK_NAME)));
}

async function main(): Promise<void> {
  try {
    const inputJson = await parseHookInput();
    const toolInput = inputJson?.tool_input ?? {};
    const command = (toolInput.command as string) ?? "";
    const sessionId = inputJson?.session_id;

    // Not a ci_monitor call - approve silently
    if (!/ci_monitor_ts\/main\.ts/.test(command)) {
      approve();
      return;
    }

    // --session-id provided (space or = separator) - approve with log
    // Pattern: --session-id=value or --session-id value
    if (/--session-id(?:=\S+|\s+\S+)/.test(command)) {
      await logHookExecution(HOOK_NAME, "approve", "--session-id provided", {}, { sessionId });
      approve();
      return;
    }

    // ci_monitor called without --session-id - warn
    const warning = `[${HOOK_NAME}] 警告: --session-id が指定されていません

ログが正しいセッションと紐付かなくなります。
--session-id を追加してください:

  bun run .claude/scripts/ci_monitor_ts/main.ts {PR} --session-id <SESSION_ID>

※ <SESSION_ID>はUserPromptSubmit hookで提供されるセッションID
  例: 3f03a042-a9ef-44a2-839a-d17badc44b0a`;

    // Include warning in systemMessage for display
    const result = {
      ...makeApproveResult(HOOK_NAME),
      systemMessage: warning,
    };
    await logHookExecution(
      HOOK_NAME,
      "approve_with_warning",
      "--session-id missing",
      {
        command: command.slice(0, 100),
      },
      { sessionId },
    );
    console.log(JSON.stringify(result));
  } catch (error) {
    // On error, approve to avoid blocking legitimate commands
    console.error(`[${HOOK_NAME}] Hook error:`, error);
    await logHookExecution(HOOK_NAME, "approve", `Hook error: ${formatError(error)}`);
    approve();
  }
}

if (import.meta.main) {
  main().catch((error) => {
    console.error(`[${HOOK_NAME}] Unexpected error:`, error);
    console.log(JSON.stringify({}));
    process.exit(0); // Don't block on error
  });
}
